assume/units/dst_components.py

```diff
@@ -2,782 +2,1593 @@
-            + b.lime_demand[t] * model.lime_price
-        )
-
-    return model_part
-
-
-def create_hydrogen_storage(
-    model,
-    max_capacity,
-    min_capacity,
-    initial_soc,
-    storage_loss_rate,
-    charge_loss_rate,
-    discharge_loss_rate,
-    time_steps,
-    **kwargs,
-):
+        return model_block
+
+
+class ElectricVehicle(GenericStorage):
     """
-    Represents a generic energy storage unit.
+    A class to represent an Electric Vehicle (EV) unit in an energy system model.
+
+    The class encapsulates the parameters, variables, and constraints necessary to model
+    the behavior of an EV, including charging and discharging, battery storage limits, availability
+    profiles, and ramp rates.
+
+    Inherits from GenericStorage and adds EV-specific functionality, such as availability profiles
+    and predefined charging profiles.
 
     Args:
-        model: A Pyomo ConcreteModel object representing the optimization model.
-        id (str): A unique identifier for the storage unit.
-        max_capacity (float): The maximum storage capacity of the unit.
-        min_capacity (float): The minimum storage capacity of the unit.
-        initial_soc (float): The initial state of charge (SOC) of the storage unit.
-        storage_loss_rate (float): The rate of energy loss due to storage inefficiency.
-        charge_loss_rate (float): The rate of energy loss during charging.
-        discharge_loss_rate (float): The rate of energy loss during discharging.
-        **kwargs: Additional keyword arguments.
-
-    Constraints:
-        storage_min_capacity_constraint: Ensures the SOC of the storage unit stays above the minimum capacity.
-        storage_max_capacity_constraint: Ensures the SOC of the storage unit stays below the maximum capacity.
-        energy_in_max_capacity_constraint: Limits the charging of the storage unit to its maximum capacity.
-        energy_out_max_capacity_constraint: Limits the discharging of the storage unit to its maximum capacity.
-        energy_in_uniformity_constraint: Ensures uniformity in charging the storage unit.
-        energy_out_uniformity_constraint: Ensures uniformity in discharging the storage unit.
-        storage_capacity_change_constraint: Defines the change in SOC of the storage unit over time.
-    """
-    model_part = pyo.Block()
-    # define parameters
-    model_part.max_capacity = pyo.Param(initialize=max_capacity)
-    model_part.min_capacity = pyo.Param(initialize=min_capacity)
-    model_part.initial_soc = pyo.Param(initialize=initial_soc)
-    model_part.storage_loss_rate = pyo.Param(initialize=storage_loss_rate)
-    model_part.charge_loss_rate = pyo.Param(initialize=charge_loss_rate)
-    model_part.discharge_loss_rate = pyo.Param(initialize=discharge_loss_rate)
-
-    # define variables
-    model_part.soc = pyo.Var(time_steps, within=pyo.NonNegativeReals)
-    model_part.uniformity_indicator = pyo.Var(time_steps, within=pyo.Binary)
```

Comment:
@nick-harder @HafnerMichael We need model_part.uniformity_indicator = pyo.Var(time_steps, within=pyo.Binary) and the corresponding constraints (which were removed) to logically enforce that storage cannot both charge and discharge simultaneously. The binary variable uniformity_indicator ensures that this behavior is prevented. Allowing the storage unit to charge and discharge simultaneously is not physically feasible and contradicts real-world operational constraints.

Reply:
@Manish-Khanra I would disagree. When the storage is configured properly, namely the charging and discharging efficiencies are specified, it would not charge and discharge at the same time as this would not be optimal behavior. Also, even if this behavior would occur for a battery storage, this would mean a simple financial flow and not physical and the final energy delivered or purchased would be the sum of both flows. If you would like, we could implement this constraint for the DRI and hydrogen storages as additional constraints

Reply:
@Manish-Khanra I have now added the additional constraint to the DRI and hydrogen storage technologies.